### ISSUE TITLE
Widen on-prem virtualisation support beyond VMWare (e.g. to Proxmox)

### ISSUE THREAD

**Issue by udf2457** [bottlerocket-os/bottlerocket#2682]
Based on a comment in #1097 

TL;DR:  See subject-line of this feature request. ;)

At present there is [bare-metal and VMWare support](https://github.com/bottlerocket-os/bottlerocket/blob/develop/PROVISIONING-METAL.md) which is good, but on the virtualisation side it would be good to reach a wider audience (VMWare comes with an enterprisey price tag attached).

Proxmox is a very popular SMB-level solution for those still looking for commercial support but not able to justify VMWare or Microsoft tier hypervisors.

Given that there are (as I understand it) technical reasons why bottlerocket cannot be shipped as an ISO file, it would be good to have some more first-class support for second-tier virtualisation.

**Comment by stmcginnis** [comment]
Thanks @udf2457!

**Comment by aRustyDev** [comment]
I'm not sure if this would count as "more detailed proposal" but is there some way that we could define some `packer` like configs, that would enable starting with bottlerocket and ending up with whatever `packer` can output?

I'm not clear on the technical reasons that BottleRocket isn't shippable as an ISO, but does that mean that its not possible to package it that way? From what I can tell, using the `packer` `qemu` integration requires an ISO as the source, but would enable outputting a `qcow2` formatted image, so proxmox could use it.

**Comment by udf2457** [comment]
If one were being cynical @aRustyDev , one might suggest that the reason they're not shipping ISOs or indeed QCOW2 files is because its an AWS project and so they don't want it being used on other cloud provider platforms (or at least not easily).  Hence the arbitrary restriction to bare metal and VMware.

I would like to think otherwise, but in the absence of any clear explanation otherwise....

**Comment by bcressey** [comment]
There are raw image files provided for the `metal-*` variants - see [instructions to fetch](https://github.com/bottlerocket-os/bottlerocket/blob/develop/PROVISIONING-METAL.md#fetch-the-bottlerocket-image-for-bare-metal).

The challenge (which the linked documentation makes painfully clear) is that bare metal takes a fair bit of legwork in order to provide the initial settings to Bottlerocket in the way it expects. If the project had an installer, that's what the installer would help walk through, but it doesn't currently.

For other virtualization environments, the better answer is going to be adding another [early-boot-config](https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/early-boot-config/user-data-providers) user-data provider, and another family of variants that combine that provider with additional packages (`kubelet`, etc).

We have a long-running effort to make adding more variants easier to do, tracked in #2669 - only slightly older than this issue! - and are close to finishing that work.

**Comment by bcressey** [comment]
This is not currently planned, but should be possible to achieve with Twoliter and the out-of-tree builds framework.
